app/src/androidTest/assets/features/publicPrivateKeyGeneration.feature

```diff
@@ -0,0 +1,15 @@
+Feature: Public and private key generation
+  As a user
+  I want to have the app generate a public/private key pair when I use it for the first time
+  So that I don't have to generate a pair myself
```

Comment:
Aren't these pairs also used later to establish a secure connection between two users. What I mean to say is maybe this is not the only reason.